swarm/network/stream/delivery.go

```diff
@@ -272,7 +279,7 @@ func (d *Delivery) RequestFromPeers(ctx context.Context, req *network.Request) (
 		Addr:      req.Addr,
 		SkipCheck: req.SkipCheck,
 		HopCount:  req.HopCount,
-	}, Top, "request.from.peers")
+	}, Top, fmt.Sprintf("request.%v.%v", sp.ID(), req.Addr))
```

Comment:
This will call `SendPriority` with a unique `traceId` on every call, resulting in, `StartSpan` being called with unique identifier every time.
```
		ctx, sp = spancontext.StartSpan(
			ctx,
			traceId,
		)
```

`StartSpan` records the `operation`, and `operation` should not be unique. We record trace-specific information in the spans via `tags`, and `logs`, but the `operation` must NOT be unique.

Considering that you must however keep track of multiple `request.from.peers` spans, you'd have to use a unique identifier to add them to your `sync.Map`, but the operation name used in `StartSpan` must stay common, such as `request.from.peers`.